source/common/filter/config_discovery_impl.h

```diff
@@ -304,14 +383,15 @@ class FilterConfigProviderManagerImplBase : Logger::Loggable<Logger::Id::filter>
 /**
  * An implementation of FilterConfigProviderManager.
  */
-template <class FactoryCb>
+template <class Factory, class FactoryCb, class FactoryCtx, class ListenerDynFilterCfg,
+          STAT_PREFIX_TYPE StatPrefixIndex>
```

Comment:
Not sure how I feel about this pattern, is there no cleaner way to do it?

Reply:
This pattern is to address an earlier comments that the difference of FilterConfigProviderManagerImpl in HTTP filter, TCP listener filter and UDP listener filter is very small. And it can be abstracted out into one base class with typed and non-typed template parameter like this to avoid duplicated code.